### PR TITLE
Implement leveling guide v1264 UIF.

### PR DESCRIPTION
**Pull request by stevewgr**
### Description

This PR is continution of @xGuTeK's work from here: https://github.com/ko4life-net/ko/pull/157

As it had multiple commits from previous PRs, I rebased it with `master` branch, as well as dropped the irrelevant commits.

I'll run and tests and apply fixes where needed before merging, therefore this PR is created as draft.

### CODE REVIEW

**Comment by stevewgr** [comment]
Note that this PR needs to be merged with this one: https://github.com/ko4life-net/ko-assets/pull/7

File: src/game/UILevelGuide.cpp [hunk 1]
```diff
@@ -236,10 +120,140 @@ bool CUILevelGuide::ReceiveMessage(CN3UIBase * pSender, DWORD dwMsg) {
...
+    CN3UIBase::SetVisible(bVisible);
+    if (bVisible) {
+        CGameProcedure::s_pUIMgr->SetVisible(this);
+    } else {
+        CGameProcedure::s_pUIMgr->ReFocusUI();
+    }
+}
+
+void CUILevelGuide::UpdateQuestsByLevel(int iLevel) {
+    if (iLevel >= 0 && CGameBase::s_pPlayer->m_InfoBase.iLevel + 5 >= iLevel) {
```
Comment (xGuTeK) [inline src/game/UILevelGuide.cpp:156]: change it to 

 if (iLevel >= 0 && CGameBase::s_pPlayer->m_InfoBase.iLevel + MAX_SEARCH_ABOVE_LEVEL >= iLevel) {

File: src/game/UILevelGuide.cpp [hunk 2]
```diff
@@ -114,10 +126,134 @@ bool CUILevelGuide::ReceiveMessage(CN3UIBase * pSender, DWORD dwMsg) {
...
+        CGameProcedure::s_pUIMgr->SetVisible(this);
+    } else {
+        CGameProcedure::s_pUIMgr->ReFocusUI();
+    }
 }
 
-void CUILevelGuide::MovePage(int iIndex) {
-    N3_WARN("[CUILevelGuide::MovePage] TODO: needs implementing.");
+void CUILevelGuide::UpdateQuestsByLevel(int iLevel) {
+    if (iLevel >= 0 && CGameBase::s_pPlayer->m_InfoBase.iLevel + 5 >= iLevel) {
```
Comment (xGuTeK) [inline src/game/UILevelGuide.cpp:156]:
```suggestion 
    if (iLevel >= 0 && CGameBase::s_pPlayer->m_InfoBase.iLevel + MAX_SEARCH_ABOVE_LEVEL  >= iLevel) {
```

File: src/game/UILevelGuide.cpp [hunk 3]
```diff
@@ -114,10 +126,134 @@ bool CUILevelGuide::ReceiveMessage(CN3UIBase * pSender, DWORD dwMsg) {
...
+
+void CUILevelGuide::CheckAvailableQuests() {
+    int iLevelFilter = std::atol(m_pEdit_Level->GetString().c_str());
+    if (iLevelFilter <= 0) {
+        return;
+    }
+
+    if (CGameBase::s_pPlayer->m_InfoBase.iLevel + MAX_SEARCH_ABOVE_LEVEL < iLevelFilter) {
+        std::string szMsgFormat;
+        ::_LoadStringFromResource(IDS_LEVELGUIDE_MAX_SEARCH_ABOVE_LEVEL, szMsgFormat);
```
Comment (xGuTeK) [inline src/game/UILevelGuide.cpp:247]:
```suggestion
        ::_LoadStringFromResource(IDS_LEVELGUIDE_SEARCH_WARNING, szMsgFormat);
```

**Review by xGuTeK — CHANGES_REQUESTED** [review]

File: src/game/res/Resource.h [hunk 4]
```diff
@@ -514,6 +514,7 @@
 #define IDS_DEAD_REVIVAL_MESSAGE                 6603
 #define IDS_DEAD_LOW_LEVEL                       6604
 #define IDS_INVEN_WEIGHT                         6605
+#define IDS_LEVELGUIDE_MAX_SEARCH_ABOVE_LEVEL    10100
```
[the remark this answers is not in the document]
Comment (stevewgr) [reply src/game/res/Resource.h]: Good point. Please see: https://github.com/ko4life-net/ko/pull/163/commits/1fa07a15b225b839975925b4841bbfa1d187fc71

File: src/game/UILevelGuide.cpp (hunk 2 above)
Comment (stevewgr) [reply src/game/UILevelGuide.cpp]: Good catch! Please see: https://github.com/ko4life-net/ko/pull/163/commits/1fa07a15b225b839975925b4841bbfa1d187fc71

File: src/game/UILevelGuide.cpp (hunk 3 above)
Comment (stevewgr) [reply src/game/UILevelGuide.cpp]: Good point! Please see: https://github.com/ko4life-net/ko/pull/163/commits/1fa07a15b225b839975925b4841bbfa1d187fc71

**Comment by stevewgr** [comment]
I applied the requested changes, rebased the commits and build is passing: https://github.com/ko4life-net/ko/actions/runs/3308706058/jobs/5461283807

Thanks @xGuTeK for the great work! Let's merge this.

**Pull request merged**